diff --git a/ext_lib/fftw/CMakeLists.txt b/ext_lib/fftw/CMakeLists.txt
index 37fbbd8..4fbf01a 100644
--- a/ext_lib/fftw/CMakeLists.txt
+++ b/ext_lib/fftw/CMakeLists.txt
@@ -21,13 +21,16 @@ else()
               http://www.fftw.org/fftw-2.1.5.tar.gz
             URL_HASH
               MD5=8d16a84f3ca02a785ef9eb36249ba433
-            CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+            CMAKE_ARGS "${CMAKE_ARGS};-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
             PREFIX ${CMAKE_CURRENT_BINARY_DIR}
             CONFIGURE_COMMAND ""
             BUILD_COMMAND ""
             # This cp line is important. FFTW 2 must be build inplace otherwise
             # the make part will generate invalid library...
             COMMAND cp -r ${CMAKE_CURRENT_BINARY_DIR}/src/fftw_external/. .
+            # Update config.sub and config.guess so fftw can be compiled with new target osx-arm64
+            COMMAND wget -O config.sub https://git.savannah.gnu.org/gitweb/?p=config.git&a=blob_plain&f=config.sub&hb=HEAD
+            COMMAND wget -O config.guess https://git.savannah.gnu.org/gitweb/?p=config.git&a=blob_plain&f=config.guess&hb=HEAD
             COMMAND ./configure --enable-float --with-pic
             COMMAND sed -i.bak -e "s/^CFLAGS = /CFLAGS = -fPIC /" Makefile
             COMMAND make -j8
diff --git a/python/setup.py b/python/setup.py
index 1c8669c..23ed2d7 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -45,6 +45,7 @@ class CMakeBuild(build_ext):
         if 'MODE' in env:
             if env['MODE'] == 'omp':
                 cmake_args += ['-DUSE_OPENMP=ON']
+        cmake_args += os.getenv('CMAKE_ARGS','').split(' ')
         if os.name == 'nt':
           # We need makefile generator that does not support multi-configuration builds
           # otherwise windows will output a Release or Debug folder in which the artifacts
